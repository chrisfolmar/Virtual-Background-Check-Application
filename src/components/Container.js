import React, { Component } from 'react'
import GoogleMapsContainer from './GoogleMapsContainer'
import DataWrapper from './DataWrapper'
import SearchInput from './SearchInput'
import superagent from 'superagent'
//import { fetchBadIp } from '../api/badIp.js'
//import { fetchGeoIp } from '../api/geoIp.js'

class Container extends Component {
    constructor(props) {
        super(props);

        this.state = {
            fieldVal: "",
            data: {},
            address: "",
            latitude: null,
            longitude: null,
            found: false,
            hostname: "",
            continent: "",
            country: "",
            region: "",
            city: "",
            postal: "",
            timeZone: "",
            asName: "",
            asNumber: "",
            asCountry: "",     
        };

        this.ipSearch = this.ipSearch.bind(this);
    }

    componentDidMount() {
        this.loadInitialData();
    }

    loadInitialData(){
        const badIpUrl = "https://api.apility.net/badip/" + process.env.REACT_APP_MY_IP + "?token=" + process.env.REACT_APP_APILITY_KEY

        /*fetchBadIp.then( (res) => {
            if(res.ok){
                    this.setState({
                        found: true
                    })
                }
            else
            {
                this.setState({
                    found: false
                })
            }
        })*/

        /*fetch(badIpUrl)
            .then( (res) => {
                if(res.ok){
                        this.setState({
                            found: true
                        })
                    }
                else
                {
                    this.setState({
                        found: false
                    })
                }
            })*/

        superagent
        .get(badIpUrl)
        .query(null)
        .set('Accept', 'text/json')
        .end((error, response) => {

            if(response.ok){

                this.setState({
                    found: true,
                    blacklisted: "The IP was found in at least one blacklist!"
                })
            }
            else
            {
                this.setState({
                    found: false,
                    blacklisted: "The IP was not found in the database of blacklists!"
                })
            }
        })


        const geoIpUrl = "https://api.apility.net/geoip/" + process.env.REACT_APP_MY_IP + "?token=" + process.env.REACT_APP_APILITY_KEY
    
        /*fetchGeoIp.then( (data) => {

                const ipData = data.ip;

                this.setState({
                    ipData: ipData,

                    address: ipData.address,

                    latitude: parseFloat(ipData.latitude),

                    longitude: parseFloat(ipData.longitude),

                    hostname: ipData.hostname,

                    continent: ipData.continent,

                    country: ipData.country,

                    region: ipData.region,

                    city: ipData.city,

                    postal: ipData.postal,

                    timeZone: ipData.time_zone,

                    asNumber: ipData.as.asn,

                    asName: ipData.as.name,

                    asCountry: ipData.as.country
                })
            })
        */

        /*fetch(geoIpUrl)
            .then( (res) => res.json() )
            .then( (data) => {

                const ipData = data.ip;

                this.setState({
                    ipData: ipData,

                    address: ipData.address,

                    latitude: parseFloat(ipData.latitude),

                    longitude: parseFloat(ipData.longitude),

                    hostname: ipData.hostname,

                    continent: ipData.continent,

                    country: ipData.country,

                    region: ipData.region,

                    city: ipData.city,

                    postal: ipData.postal,

                    timeZone: ipData.time_zone,

                    asNumber: ipData.as.asn,

                    asName: ipData.as.name,

                    asCountry: ipData.as.country
                })
            })
        */

        superagent
        .get(geoIpUrl)
        .query(null)
        .set('Accept', 'text/json')
        .end((error, response) => {

            const data = response.body.ip

            this.setState({
                data: data,

                address: data.address,

                latitude: parseFloat(data.latitude),

                longitude: parseFloat(data.longitude),

                hostname: data.hostname,

                continent: data.continent,

                country: data.country,

                region: data.region,

                city: data.city,

                postal: data.postal,

                timeZone: data.time_zone,

                asNumber: data.as.asn,

                asName: data.as.name,

                asCountry: data.as.country
            })
        })
    }

    ipSearch(val){
        const ipSearchBadIpUrl = "https://api.apility.net/badip/" + val + "?token=" + process.env.REACT_APP_APILITY_KEY
            
        fetch(ipSearchBadIpUrl)
            .then( (res) => {

                if(res.ok){
                    this.setState({
                        found: true
                    })
                }
                else
                {
                    this.setState({
                        found: false
                    })
                }

            })

        const ipSearchGeoIpUrl = "https://api.apility.net/geoip/" + val + "?token=" + process.env.REACT_APP_APILITY_KEY

        fetch(ipSearchGeoIpUrl)
            .then( (res) => res.json() )
            .then( (data) => {

                const searchData = data.ip;

                this.setState({
                    ipData: searchData,

                    address: searchData.address,

                    latitude: parseFloat(searchData.latitude),

                    longitude: parseFloat(searchData.longitude),

                    hostname: searchData.hostname,

                    continent: searchData.continent,

                    country: searchData.country,

                    region: searchData.region,

                    city: searchData.city,

                    postal: searchData.postal,

                    timeZone: searchData.time_zone,

                    asNumber: searchData.as.asn,

                    asName: searchData.as.name,

                    asCountry: searchData.as.country
                })
            })
    }

    render (){

        const center = {
            lat: this.state.latitude,
            lng: this.state.longitude
        }

        return (
            <div>
                <SearchInput onSubmission={this.ipSearch} />

                <div style={{width: '100%', height: '30vh', background: 'grey'}}>
                    <GoogleMapsContainer style={{position: 'relative'}} center={center} zoom={10} />
                </div>

                <DataWrapper
                    found={this.state.found}
                    address={this.state.address}
                    hostname={this.state.hostname}
                    continenet={this.state.continent}
                    country={this.state.country}
                    region={this.state.region}
                    city={this.state.city}
                    postal={this.state.postal}
                    longitude={this.state.longitude} 
                    latitude={this.state.latitude}
                    timeZone={this.state.timeZone} 
                    asNumber={this.state.asNumber}
                    asName={this.state.asName}
                    asCountry={this.state.asCountry}
                />
            </div>
        )
    }
}

export default Container;
