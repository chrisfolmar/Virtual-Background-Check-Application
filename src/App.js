import React, { Component } from 'react';
import Container from './components/Container.js'

class App extends Component {
  render() {

    return (
      <div>
        <Container />
      </div>
    );
  }
}

export default App;
