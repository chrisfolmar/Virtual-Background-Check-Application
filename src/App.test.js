import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import App from './App';
import Container from './components/Container';

describe('App', () => {
	it('renders without crashing', () => {
		shallow(<App />);
	});

	it('renders Container component', () => {
		const component = shallow(<App />);
		const container = <Container />;
		// expect (wrapper.contains(container)).to.equal(true);
		expect(component.contains(container)).toEqual(true);
	});

	it('full snapshot rendering', () => {
		const component = mount(
			<App />
			);
		expect(mountToJson(component)).toMatchSnapshot();
		component.unmount();
	});
});