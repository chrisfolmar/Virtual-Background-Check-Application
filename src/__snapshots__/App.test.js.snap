// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App full snapshot rendering 1`] = `
<App>
  <div>
    <Container>
      <div>
        <SearchInput
          onSubmission={[Function]}
        >
          <header
            className="masthead text-white text-center"
          >
            <div
              className="overlay"
            />
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-xl-9 mx-auto"
                >
                  <h2
                    className="mb-5"
                  >
                    Verify if an IP Address is 
                    <span
                      style={
                        Object {
                          "color": "black",
                        }
                      }
                    >
                      blacklisted
                    </span>
                  </h2>
                </div>
                <div
                  className="col-md-10 col-lg-8 col-xl-7 mx-auto"
                >
                  <form
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-row"
                    >
                      <div
                        className="col-12 col-md-9 mb-2 mb-md-0"
                      >
                        <input
                          className="form-control form-control-lg"
                          onChange={[Function]}
                          pattern="\\\\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\\\.|$)){4}\\\\b"
                          placeholder="Enter IP Address to verify..."
                          title="1.2.3.4"
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        className="col-12 col-md-3"
                      >
                        <button
                          className="btn btn-block btn-lg btn-primary"
                          type="submit"
                        >
                          Search
                        </button>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </header>
        </SearchInput>
        <div
          style={
            Object {
              "background": "grey",
              "height": "30vh",
              "width": "100%",
            }
          }
        >
          <Wrapper
            center={
              Object {
                "lat": null,
                "lng": null,
              }
            }
            zoom={10}
          >
            <DefaultLoadingContainer>
              <div>
                Loading...
              </div>
            </DefaultLoadingContainer>
          </Wrapper>
        </div>
        <DataWrapper
          address=""
          asCountry=""
          asName=""
          asNumber=""
          city=""
          continenet=""
          country=""
          found={false}
          hostname=""
          latitude={null}
          longitude={null}
          postal=""
          region=""
          timeZone=""
        >
          <section
            className="features-icons bg-light text-center"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <GoodIp
                  address=""
                >
                  <div
                    className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                  >
                    <h4 />
                    <div
                      className="features-icons-icon d-flex"
                    >
                      <i
                        className="fa fa-check-circle-o m-auto"
                        style={
                          Object {
                            "color": "#06B400",
                          }
                        }
                      />
                    </div>
                    <h5
                      style={
                        Object {
                          "color": "#06B400",
                        }
                      }
                    >
                      Not Found!
                    </h5>
                    <p
                      className="mb-0"
                    >
                      The IP was not found in the database of blacklists!
                    </p>
                  </div>
                </GoodIp>
              </div>
              <h4>
                Geolocation Data
              </h4>
              <div
                className="row"
              >
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="Hostname"
                    icon="icon-list"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-list m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Hostname
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    header="Continent"
                    icon="fa fa-globe"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="fa fa-globe m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Continent
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="Country"
                    icon="icon-globe-alt"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-globe-alt m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Country
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="Region"
                    icon="icon-direction"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-direction m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Region
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="City"
                    icon="fa fa-building-o"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="fa fa-building-o m-auto text-primary"
                        />
                      </div>
                      <h5>
                        City
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="Postal Code"
                    icon="icon-envelope"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-envelope m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Postal Code
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    data="null, null"
                    header="Latitude, Longitude"
                    icon="icon-map"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-map m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Latitude, Longitude
                      </h5>
                      <p
                        className="mb-0"
                      >
                        null, null
                      </p>
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-3"
                >
                  <Data
                    data=""
                    header="Time Zone"
                    icon="icon-clock"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-clock m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Time Zone
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
              </div>
              <h4>
                Autonomous System Information
              </h4>
              <div
                className="row"
              >
                <div
                  className="col-lg-4"
                >
                  <Data
                    data=""
                    header="Number"
                    icon="icon-credit-card"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-credit-card m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Number
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-4"
                >
                  <Data
                    data=""
                    header="Name"
                    icon="icon-screen-desktop"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-screen-desktop m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Name
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
                <div
                  className="col-lg-4"
                >
                  <Data
                    data=""
                    header="Country"
                    icon="icon-globe-alt"
                  >
                    <div
                      className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
                    >
                      <div
                        className="features-icons-icon d-flex"
                      >
                        <i
                          className="icon-globe-alt m-auto text-primary"
                        />
                      </div>
                      <h5>
                        Country
                      </h5>
                      <p
                        className="mb-0"
                      />
                    </div>
                  </Data>
                </div>
              </div>
            </div>
          </section>
        </DataWrapper>
      </div>
    </Container>
  </div>
</App>
`;
